### ISSUE TITLE
Swift error in 10.6

### ISSUE THREAD

**Issue by hrkm** [firebase/firebase-ios-sdk#10966]
### Description

https://stackoverflow.com/questions/75704453/firebasesessions-swift-error-with-xcode-12-5-1/75773723#75773723

After upgrading from 10.3 to 10.6 now I'm getting a casting error in my XCode. I've created this simple update to the function to add the casting and it compiles fine. The minor difference is that I casted it to non-optional type (e.g. missing the "?"), so can update the pull request to have that. I'm not sure why Swift is complaining about this cast, but updating the code as in the pull request let me build my project. Please note that the function seems to be for debugging purposes only, it is not required for actual logic of Firebase.

https://github.com/firebase/firebase-ios-sdk/compare/master...hrkm:firebase-ios-sdk:patch-1#diff-17a6d07d68bd3d24b12ede65f118e609eecec47c8df92c5afb9d22199712707f

### Firebase SDK Version

10.6

### Xcode Version

13.2.1

### Installation Method

CocoaPods

### Firebase Product(s)

Analytics, Authentication, Crashlytics, Firestore

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by paulb777** [comment]
We've tested Firebase 10.6 code successfully on the earliest supported Xcode version - 13.3.1.  Please upgrade to at least that Xcode version.  Also, note that Apple will soon be dropping Xcode 13 support and Firebase will follow and move to require at least 14.1.

**Comment by hrkm** [comment]
Yeah, I kind of realized that probably the error might be related to the old version of Xcode, especially after submitting my app bundle to AppStore which also displayed the warning that Xcode 14 will be required starting April 2023. [which started showing up since yesterday - I had uploaded another package on March 15th and didn't see that warning at all - that's a very small amount of time to upgrade, but that's on Apple :( ]

The patch I made is useful for anyone who still need to upgrade but needs to release their new bundle before getting new equipment (for me it means I need to get a new device). Feel free to close the issue, but hopefully others will be able to find it if they encounter the same error in the next 2-3 weeks before Xcode 13 gets deprecated.
